falcon/request.py

```diff
@@ -1096,6 +1112,20 @@ def _parse_form_urlencoded(self):
 
             self._params.update(extra_params)
 
+    def _parse_application_json(self):
+        raw_body = self.stream.read()
+
+        body = None
+        try:
+            body = json.loads(raw_body.decode('utf-8'))
```

Comment:
Instead of hard-coding as "utf-8", shouldn't this follows [HTTP charset parameter](http://www.w3.org/International/O-HTTP-charset)?